### PR TITLE
JBPM-5409: More bug fixing and unit tests.

### PR DESCRIPTION
**Pull request by romartin**
Hello @manstis , @jlindop ,
Here is the second iteration of cleans, bug fixes and adding more tests. Could you please review it?
Details on [this ticket](https://issues.jboss.org/browse/JBPM-5409)
PS: @jlindop is this gets merged before yours, you'll have to rebase, hope no conflicts or at least diffcult conflicts appears, if any issues just lemme know!
Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
Added new commit

**Comment by romartin** [comment]
jenkins please retest this

**Comment by jlindop** [comment]
Tested and the editor is looking better and better.
+1

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/command/CanvasCommandFactory.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    protected CanvasCommandFactory() {
+        this.treeWalkTraverseProcessor = null;
+    }
+
+    @Inject
+    public CanvasCommandFactory( final TreeWalkTraverseProcessor treeWalkTraverseProcessor ) {
+        this.treeWalkTraverseProcessor = treeWalkTraverseProcessor;
+    }
+
+    public AddNodeCommand ADD_NODE( Node candidate, String shapeSetId ) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/command/CanvasCommandFactory.java]: hehe good point!! long story... those were initialilly static methods and .... willl refactor this, yeah well seen :)

**Pull request merged**